cmd/ctl/pkg/util/util.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2020 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	apiutil "github.com/jetstack/cert-manager/pkg/api/util"
+	cmapiv1alpha2 "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha2"
+	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+	cmclient "github.com/jetstack/cert-manager/pkg/client/clientset/versioned"
+)
+
+// Fetches the x509 certificate from a CR and stores the certificate in file specified by certFilename
+// Assumes CR is ready, otherwise returns error
+func FetchCertificateFromCR(cmClient cmclient.Interface,
+	crName, crNamespace, certFileName string,
+	ioStreams genericclioptions.IOStreams) error {
+	req, err := cmClient.CertmanagerV1alpha2().CertificateRequests(crNamespace).Get(context.TODO(), crName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error when querying CertificateRequest: %v", err)
```

Comment:
Haven't tested this but a `%w` could be better, this is quite new in go but allows better error handling in the future